netconf/transport_ssh.go

```diff
@@ -117,6 +122,17 @@ func NewSSHSession(conn net.Conn, config *ssh.ClientConfig) (*Session, error) {
 	return NewSession(t), nil
 }
 
+// NewSSHSessionForExternalClient creates a new NETCONF session using an existing ssh.Client
+// initiated and managed externally.
+func NewSSHSessionForExternalClient(client *ssh.Client) (*Session, error) {
```

Comment:
This name is a bit much

NewSSHSession() is fine.  The arguments show that this is externally managed.

Reply:
Will rethink the naming.